app/scripts/modules/core/src/diffs/ViewChangesLink.tsx

```diff
@@ -1,8 +1,78 @@
-import { IViewChangesConfig } from './viewChangesLink.component';
+import * as React from 'react';
+import { IBuildDiffInfo, ICreationMetadataTag, IExecution, IExecutionStage } from 'core/domain';
+import { LabeledValue, showModal, useData } from 'core/presentation';
+import { ReactInjector } from 'core/reactShims';
+import { ChangesModal } from './ChangesModal';
+import { ICommit } from './CommitHistory';
+import { IJarDiff } from './JarDiffs';
+
+export interface IViewChangesConfig {
+  buildInfo?: IBuildDiffInfo;
+  commits?: ICommit[];
+  jarDiffs?: IJarDiff;
+  metadata?: ICreationMetadataTag;
+}
 
 export interface IViewChangesLinkProps {
   changeConfig: IViewChangesConfig;
-  viewType?: string;
   linkText?: string;
   nameItem: { name: string };
+  viewType?: string;
 }
+
+export const ViewChangesLink = ({ changeConfig, linkText, nameItem, viewType }: IViewChangesLinkProps) => {
+  const changeConfigValue = changeConfig.metadata.value;
+  const isExecution = changeConfigValue.executionType === 'pipeline';
+
+  const fetchExecution = () => {
+    if (isExecution) {
+      return ReactInjector.executionService.getExecution(changeConfigValue.executionId);
+    }
+    /** A noop promise so `useData` can be utilized */
+    return (new Promise(() => {}) as unknown) as PromiseLike<IExecution>;
```

Comment:
this will return a pending promise that never resolves.  Would it be better to use `Promise.resolve({})` instead?
```suggestion
    return Promise.resolve({}) as PromiseLike<IExecution>;
```